### ISSUE TITLE
Pydantic 1.10.7 breaks with the latest release of typing-extensions (4.6.0)

### ISSUE THREAD

**Issue by c0dearm** [pydantic/pydantic#5824]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

On May 22, there was a new release of `typing-extensions`, [version 4.6.0](https://github.com/python/typing_extensions/blob/main/CHANGELOG.md#release-460-may-22-2023), which has significant changes and breaks new installations of `Pydantic`, since the requirement does not have a upper version bound for `typing-extensions`.

```
  File "pydantic/main.py", line 197, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/fields.py", line 506, in pydantic.fields.ModelField.infer
  File "pydantic/fields.py", line 436, in pydantic.fields.ModelField.__init__
  File "pydantic/fields.py", line 552, in pydantic.fields.ModelField.prepare
  File "pydantic/fields.py", line 668, in pydantic.fields.ModelField._type_analysis
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/typing.py", line 774, in __subclasscheck__
    return issubclass(cls, self.__origin__)
TypeError: issubclass() arg 1 must be a class
```

Reverting back to the previous `typing-extensions` version (4.5.0) fixes the issue.

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.7
            pydantic compiled: True
                 install path: /<omitted>/.venv/lib/python3.8/site-packages/pydantic
               python version: 3.8.16 (default, Dec  7 2022, 01:12:06)  [GCC 11.3.0]
                     platform: Linux-5.19.0-41-generic-x86_64-with-glibc2.35
     optional deps. installed: ['typing-extensions']
```
```


### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [X] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by samuelcolvin** [comment]
@hramezani please could you have a look at this.

**Comment by samuelcolvin** [comment]
Duplicate of #5821 I guess.

**Comment by bsoptei** [comment]
+1

**Comment by antazoey** [comment]
I don't see any linked PRs, anyone know specifically what the issues were?

**Comment by hramezani** [comment]
As mentioned above, This is a duplicate of https://github.com/pydantic/pydantic/issues/5821.

We already merged a fix and will prepare a new release soon

**Comment by bonastreyair** [comment]
It seems like with the "fixed" latest version of `pydantic==1.10.8` and `typing-extensions==4.6.1` it is still happening...
